### PR TITLE
Add per schema permissions.

### PR DESCRIPTION
**Pull request by bkyryliuk**
Adds per schema permissions for the user.
It will be shared among superset explore and sql lab functionality.

Reviewers:
* @mistercrunch 
* @vera-liu 
* @ascott

### CODE REVIEW

File: superset/models.py [hunk 1]
```diff
@@ -1626,6 +1630,11 @@ def num_cols(self):
     def name(self):
         return self.datasource_name
 
+    @property
+    def schema_perm(self):
+        """Is not used. Created for consistency with SqlaTable."""
```
Comment (mistercrunch) [inline superset/models.py]: I've seen people using dot notation in their druid datasource names, maybe we could try to split on `.` and take the first part of the string? Not a real requirement though.

File: superset/models.py [hunk 2]
```diff
@@ -908,6 +908,10 @@ def link(self):
         return Markup(
             '<a href="{self.explore_url}">{table_name}</a>'.format(**locals()))
 
+    @property
+    def schema_perm(self):
+        return "[{obj.database}].[{obj.schema}]".format(obj=self)
```
Comment (mistercrunch) [inline superset/models.py]: what if schema is not defined?

File: superset/security.py [hunk 3]
```diff
@@ -186,6 +186,7 @@ def sync_role_definitions():
     for datasource in datasources:
         perm = datasource.get_perm()
         sm.add_permission_view_menu('datasource_access', perm)
+        sm.add_permission_view_menu('schema_access', datasource.schema_perm)
```
Comment (mistercrunch) [inline superset/security.py]: what if schema is not defined/empty?

**Comment by mistercrunch** [comment]
LGTM

**Comment by bkyryliuk** [comment]
thanks !

**Pull request merged**